algviz/picture/connection_map.py

```diff
@@ -0,0 +1,136 @@
+import svgutils.transform as svgutils
+import os
+
+from algviz.parser import structures
+from .anchor import Anchor
+from .svg_engine import SVGEngine
+
+"""Tracks connection points and connection pictures to be composed into a single picture"""
+
+pointers_temp_file = "temp_pointers.svg"
+
+class ConnectionMap:
+    class Connection:
+        def __init__(self, start_point, connect_pic, anchor):
+            self.start_point = start_point
+            self.connect_pic = connect_pic
+            if not connect_pic.is_drawn:
+                connect_pic.draw()
+            self.anchor = anchor
+
+            self.left_shift_y = 0
+            self.right_shift_y = 0
+
+        def is_pointer(self):
+            return False
+
+    class PointerConnection(Connection):
+        def __init__(self, start_point, connect_pic, anchor, pointer_style={}):
+            self.pointer_style = pointer_style
+            super().__init__(self, start_point, connect_pic, anchor)
+
+        def is_pointer(self):
+            return True
+    
+    def __init__(self, startPicture):
+        self.connections = []
+        self.picture = startPicture
+        
+    def add_connection(self, point, connect_pic, anchor=None, pointer=True, pointer_style={}):
+        if self.picture.width < point[0] or self.picture.height < point[1]:
+            raise TypeError("Point not included in picture")
+        if pointer:
+            self.connections.append(ConnectionMap.PointerConnection(point, connect_pic, anchor, pointer_style))
+        else:
+            self.connections.append(ConnectionMap.Connection(point, connect_pic, anchor))
+
+    def _resize_pic(self):
+        left_width = 0
+        right_width = 0
+        left_height = 0
+        right_height = 0
+        for connection in self.connections:
+            if not connection.is_pointer():
+                #in place connections dont change size of picture
+                continue
+            connect_size = connection.connect_pic.size
+            if connection.anchor == Anchor.LEFT:
```

Comment:
I thought the point of using tuples of numeric values for `Anchor` attributes would be to do a little arithmetic in places like this instead of having separate code for each `Anchor` possibility.